lib/dangermattic/plugins/common/common_release_checks.rb

```diff
@@ -0,0 +1,134 @@
+    #   Example: `file_comparison = ->(file_path) { file_path.include?('app/') }`
+    #
+    # @param message [String] The message to display in the warning or failure output if the condition is met.
+    #
+    # @param on_release [Boolean] If true, the check will only run on release branches, otherwise on non-release branches.
+    #
+    # @param fail_on_error [Boolean] If true, a failure message will be displayed instead of a warning.
+    #
+    # @example Check if any modified file is under the 'app/' directory and emit a warning on release branches:
+    #   check_file_changed(file_comparison: ->(file_path) { file_path.include?('app/') },
+    #                      message: 'Some files in the "app/" directory have been modified. Please review the changes.',
+    #                      on_release: true
+    #   )
+    #
+    # @example Check if a specific file has been modified and emit a failure on non-release branches:
+    #   check_file_changed(file_comparison: ->(file_path) { file_path == 'path/to/file/DoNotChange.java' },
+    #                      message: 'The "DoNotChange.java" file has been modified. This change is not allowed on non-release branches.',
+    #                      on_release: false,
+    #                      fail_on_error: true)
+    #
+    # @return [void]
+    def check_file_changed(file_comparison:, message:, on_release:, fail_on_error: false)
+      has_modified_file = all_modified_files.any?(&file_comparison)
+
+      should_be_changed = on_release ? release_branch? : !release_branch?
+      return unless should_be_changed && has_modified_file
+
+      if fail_on_error
+        failure(message)
+      else
+        warn(message)
+      end
+    end
+
+    # Check if the release notes and store strings files are correctly updated after a modification to the release notes.
+    #
+    # @param release_notes_file [String] The name of the release notes file that should be checked for modifications.
+    #   Example: 'metadata/release_notes.txt'
+    #
+    # @param store_strings_file [String] The name of the store strings file that should be checked for modifications.
+    #   Example: 'metadata/PlayStoreStrings.po'
+    #
+    # @param fail_on_error [Boolean] If true, a failure message will be emitted instead of a warning.
+    #
+    # @example Check if the release notes file 'release_notes.txt' is modified, and the 'PlayStoreStrings.po' file is not modified:
+    #   check_release_notes_and_store_strings(release_notes_file: 'release_notes.txt', store_strings_file: 'PlayStoreStrings.po', fail_on_error: false)
+    #
+    # @return [void]
+    def check_release_notes_and_store_strings(release_notes_file:, store_strings_file:, fail_on_error: false)
+      has_modified_release_notes = danger.git.modified_files.any? { |f| f.end_with?(release_notes_file) }
+      has_modified_app_store_strings = danger.git.modified_files.any? { |f| f.end_with?(store_strings_file) }
```

Comment:
Better check the `File.basename` rather than `f.end_with?` because otherwise `f = 'previous_release_notes.txt'` would match `end_with?('release_notes.txt')` for example.

Reply:
Though at the same time, in this case given I'd expect users to specify a path, one could specify `metadata/release_notes.txt` and it would match `old/release_notes.txt` 🤔

Reply:
Indeed. So better just `==` the two strings and expect the caller to provide the _exact_ path relative to the root of the client project's repo.

Reply:
Btw, `store_strings_file` isn't super explicit and fitting of a variable name. I'd instead suggest `po_file` or something similar?

Reply:
Thanks for the suggestions -- I have changed a few things on this method here: e687615ca59e7c026dd12188cec5df1413139559